tools/actions/composites/adjust-changeset-level/action.yml

```diff
@@ -0,0 +1,34 @@
+name: Adjust Changeset Level
+description: This action adjusts the changeset level from one value to another.
+inputs:
+  from_level:
+    description: 'The changeset level to target'
+    required: true
+  to_level:
+    description: 'The desired changeset level for the target'
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Validate input levels
+      run: |
+        valid_levels=("patch" "minor" "major")
+        if [[ ! " ${valid_levels[@]} " =~ " ${inputs.from_level} " ]]; then
+          echo "Invalid from_level: ${inputs.from_level}. Must be one of ${valid_levels[@]}"
+          exit 1
+        fi
+        if [[ ! " ${valid_levels[@]} " =~ " ${inputs.to_level} " ]]; then
+          echo "Invalid to_level: ${inputs.to_level}. Must be one of ${valid_levels[@]}"
+          exit 1
+        fi
+
+    - name: Adjust changeset level
```

Comment:
Inviting comments on this. As you can see this is done by direct manipulation of the changeset files via `sed` in a shell script. `sed` targets the YAML front matter specifically and not the rest of the file, but this approach still makes the assumption that the format of the YAML front matter in the changeset file will not change to use the `level` in some different context. I think this is a very unlikely scenario and that this approach is fine for a solution that is not intended to last many years.

An alternative to is to do this in node, convert the YAML front matter into JS objects via a standard yaml parser and writer - this is the way the changesets library itself manipulates its changeset files and is more future-proof, but is slightly more code.